src/frontend/model/scaninput_manager.rs

```diff
@@ -0,0 +1,21 @@
+use leptos::prelude::WindowListenerHandle;
+
+pub const SECONDS_UNTIL_INPUT_CLEARED: u8 = 30;
+
+pub struct ScanInputManager {
```

Comment:
Could this also just be a vec which we check for emptyness?

Reply:
Well technically the option is only there, because `WindowListenerHandle`.remove() requires a owned value, so I'm abusing `Option`.take() to get an owned value out ouf a `&mut`.